doc/changelog.md

```diff
@@ -16,9 +16,28 @@
   load quantum devices dynamically.
   [(#343)](https://github.com/PennyLaneAI/catalyst/pull/343)
 
+* Jax dynamic API is enabled and the intial support for dynamically-shaped arrays is added.
+  Catalyst now accepts tensors whose dimensions are not known at the compile time in the top-level
+  programs. Standard tensor initialisation functions `jax.numpy.ones/zeros/empty` now accept
+  variables in shapes. Support for dynamic arrays in control-flow primitives is yet to be done.
+  [(#366)](https://github.com/PennyLaneAI/catalyst/pull/366)
```

Comment:
```suggestion
* Support for dynamically-shaped arrays has been added.
  [(#366)](https://github.com/PennyLaneAI/catalyst/pull/366)

  Catalyst now accepts tensors whose dimensions are not known at compile time.
  Standard tensor initialisation functions `jax.numpy.ones`, `jnp.zeros`, and
  `jnp.empty`, now accept dynamic variables (where the value is only known at
  runtime). Note that support for dynamic arrays in control-flow primitives
  is not yet supported and still a work-in-progress.
```

slightly re-worded to avoid burying the lede :)